roles/sro_deploy_custom_commit/scripts/deploy_operator.sh

```diff
@@ -0,0 +1,108 @@
+#! /bin/bash
+
+set -Eeuo pipefail
+
+set -x
+if [ "$#" -lt 1 -o "$#" -gt 4 ]; then
+    echo "Usage:"
+    echo "  $0 deploy <gpu_operator_git_repo> <gpu_operator_git_ref> <gpu_operator_image_tag>"
+    echo "  $0 undeploy"
+    exit 1
+fi
+
+script_name="$0"
+action="$1"
+shift
+
+
+if [ "${ARTIFACT_DIR:-}" ]; then
+    EXTRA_LOGS_DIR="${ARTIFACT_DIR}/$(date +%H%M%S)__sro_deploy_operator"
+    mkdir -p "${EXTRA_LOGS_DIR}"
+    echo "Using $EXTRA_LOGS_DIR to store helm logs."
+    exec > "${EXTRA_LOGS_DIR}/_sro_deploy_operator.log"
+    exec 2>&1
+else
+    EXTRA_LOGS_DIR=""
+    echo "ARTIFACT_DIR not set, not storing helm log artifacts."
+fi
+
+
+# https://stackoverflow.com/a/21189044/341106
+function parse_yaml {
+   local prefix=$2
+   local s='[[:space:]]*' w='[a-zA-Z0-9_]*' fs=$(echo @|tr @ '\034')
+   sed -ne "s|^\($s\):|\1|" \
+        -e "s|^\($s\)\($w\)$s:$s[\"']\(.*\)[\"']$s\$|\1$fs\2$fs\3|p" \
+        -e "s|^\($s\)\($w\)$s:$s\(.*\)$s\$|\1$fs\2$fs\3|p"  $1 |
+   awk -F$fs '{
+      indent = length($1)/2;
+      vname[indent] = $2;
+      for (i in vname) {if (i > indent) {delete vname[i]}}
+      if (length($3) > 0) {
+         vn=""; for (i=0; i<indent; i++) {vn=(vn)(vname[i])("_")}
+         printf("%s%s%s=\"%s\"\n", "'$prefix'",vn, $2, $3);
+      }
+   }'
+}
+
+deploy_from_helm() {
+    #TODO if/when we add helm deployment for SRO
+    echo "deploy_from_helm() currently unimplemented for SRO"
+}
+
+deploy_from_commit() {
+    if [ ! "$#" -eq 3 ]; then
+        echo "Usage: $script_name deploy_from_commit <sro_git_repo> <sro_git_ref> <sro_image_tag>"
+        exit 1
+    fi
+
+    OPERATOR_GIT_REPO="$1"
+    OPERATOR_GIT_REF="$2"
+    OPERATOR_IMAGE_TAG="$3"
+
+    cd /tmp
+    rm -rf special-resource-operator
+    git clone $OPERATOR_GIT_REPO -b $OPERATOR_GIT_REF --depth 1 special-resource-operator
+
+    cd special-resource-operator
+
+    git show --quiet
+    echo
+
+    if [ "${EXTRA_LOGS_DIR}" ]; then
+        git show --quiet > "${EXTRA_LOGS_DIR}/special-resource-operator.commit"
+    fi
+
+    TAG=$OPERATOR_IMAGE_TAG make deploy
+}
+
+undeploy() {
+    cd /tmp/special-resource-operator
+    make undeploy
+}
```

Comment:
I'm wondering if you should assume that `/tmp/special-resource-operator` already contains the SRO repo or not ...
maybe test if it exists, if not cloning, so that you're always able to undeploy no matter the state of the system